mne/io/curry/curry.py

```diff
@@ -0,0 +1,315 @@
+                  "unit": unit_dict[key],
+                  "kind": FIFFV_CHANTYPES[key]}
+            if key in ("meg", "eeg"):
+                loc = sensors["SENSORS" + CHANTYPES[key]][ind]
+                ch["loc"] = np.array(loc, dtype=float)
+
+            all_chans.append(ch)
+
+    ch_names = [chan["ch_name"] for chan in all_chans]
+    info = create_info(ch_names, sfreq)
+
+    for ind, ch_dict in enumerate(info["chs"]):
+        ch_dict["kind"] = all_chans[ind]["kind"]
+        ch_dict['unit'] = SI_UNITS[all_chans[ind]['unit'][1]]
+        ch_dict['cal'] = SI_UNIT_SCALE[all_chans[ind]['unit'][0]]
+        if ch_dict["kind"] in (FIFF.FIFFV_MEG_CH,
+                               FIFF.FIFFV_EEG_CH):
+            ch_dict["loc"] = all_chans[ind]["loc"]
+
+    return info, n_samples, is_ascii
+
+
+def _read_events_curry(fname):
+    """Read events from Curry event files.
+
+    Parameters
+    ----------
+    fname : str
+        Path to a curry event file with extensions .cef, .ceo,
+        .cdt.cef, or .cdt.ceo
+
+    Returns
+    -------
+    events : ndarray, shape (n_events, 3)
+        The array of events.
+    """
+    check_fname(fname, 'curry event', ('.cef', '.cdt.cef'),
+                endings_err=('.cef', '.cdt.cef'))
+
+    events_dict = _read_curry_lines(fname, ["NUMBER_LIST"])
+    # The first 3 column seem to contain the event information
+    curry_events = np.array(events_dict["NUMBER_LIST"], dtype=int)[:, 0:3]
+
+    return curry_events
+
+
+def _read_annotations_curry(fname, sfreq='auto'):
+    r"""Read events from Curry event files.
+
+    Parameters
+    ----------
+    fname : str
+        The filename.
+    sfreq : float | 'auto'
+        The sampling frequency in the file. If set to 'auto' then the
+        ``sfreq`` is taken from the respective info file of the same name with
+        according file extension (\*.dap for Curry 7; \*.cdt.dpa for Curry8).
+         So data.cef looks in data.dap and data.cdt.cef looks in data.cdt.dpa.
+
+    Returns
+    -------
+    annot : instance of Annotations | None
+        The annotations.
+    """
+    events = _read_events_curry(fname)
+
+    if sfreq == 'auto':
+        fname_base, ext = fname.split(".", maxsplit=1)
+        curry_vers = _get_curry_version(ext)
+        with open(fname_base + INFO_FILE_EXTENSION[curry_vers]) as fid:
+            for line in fid:
+                if ('SampleFreqHz' or 'SAMPLE_FREQ_HZ') in line:
+                    sfreq = float(line.split("=")[1])
+
+    onset = events[:, 0] / sfreq
+    duration = np.zeros(events.shape[0])
+    description = events[:, 2]
+
+    return Annotations(onset, duration, description)
+
+
+def read_raw_curry(input_fname, preload=False, verbose=None):
+    """Read raw data from Curry files.
+
+    Parameters
+    ----------
+    input_fname : str
+        Path to a curry file with extensions .dat, .dap, .rs3, .cdt, cdt.dpa,
+        .cdt.cef or .cef.
+    preload : bool or str (default False)
```

Comment:
the description of preload param should be in doc_dict and common to all readers.

to be done in a next PR though

Reply:
this is needed before merging

Reply:
I added 'preload' to docdict and adapted curry docstrings to it.
I'd say adjusting the other IO docstrings should be done in another PR.